ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/AndroidApplicationContext.kt

```diff
@@ -13,20 +14,31 @@ internal interface ApplicationContext {
     fun getLocale(): Locale
 }
 
-internal class AndroidApplicationContext(private val applicationContext: Context) : ApplicationContext {
+internal class AndroidApplicationContext(
+    private val applicationName: String,
+    private val resources: Resources,
+) : ApplicationContext {
+    constructor(
+        context: Context,
+    ) : this(context.applicationContext.applicationName(), context.resources)
+
     override fun getApplicationName(): String {
-        return applicationContext.applicationInfo.loadLabel(applicationContext.packageManager).toString()
+        return applicationName
     }
 
     override fun getString(@StringRes resId: Int, vararg formatArgs: Any): String {
-        return applicationContext.resources.getString(resId, *formatArgs)
+        return resources.getString(resId, *formatArgs)
     }
 
     override fun getLocale(): Locale {
-        return applicationContext.resources.configuration.locales.get(0)
+        return resources.configuration.locales.get(0)
     }
 }
 
 internal fun Context.toAndroidContext(): ApplicationContext {
```

Comment:
If we rename the above, we will also need to rename a few things like this extension function.